public/app/features/dashboard-scene/settings/version-history/utils.ts

```diff
@@ -0,0 +1,100 @@
+import { compare, Operation } from 'fast-json-patch';
+// @ts-ignore
+import jsonMap from 'json-source-map';
+import { flow, get, isArray, isEmpty, last, sortBy, tail, toNumber, isNaN } from 'lodash';
+
+export type Diff = {
+  op: 'add' | 'replace' | 'remove' | 'copy' | 'test' | '_get' | 'move';
+  value: unknown;
+  originalValue: unknown;
+  path: string[];
+  startLineNumber: number;
+};
+
+export type Diffs = {
+  [key: string]: Diff[];
+};
+
+export const jsonDiff = (lhs: any, rhs: any): Diffs => {
```

Comment:
Can we try to create a JSON type for these two? Something like:
```suggestion
type JSONValue =
    | string
    | number
    | boolean
    | { [x: string]: JSONValue }
    | Array<JSONValue>;

export const jsonDiff = (lhs: JSONValue, rhs: JSONValue): Diffs => {
```

Reply:
This is a good call, thank you for pointing it out! I'll make the modifications!